lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -125,9 +128,16 @@ public System.Threading.Tasks.Task RunTask()
 
                         if (Logger.CustomLevel >= resultLogLevel && result != null && result.Length > 0)
                         {
-                            Logger.Log(resultLogLevel, "Task running result:\r\n" + System.Text.Encoding.Default.GetString(result));
+                            Logger.Log(resultLogLevel,
+                                "Task running result:\r\n" + System.Text.Encoding.Default.GetString(result));
                         }
                     }
+                    catch (Exception)
+                    {
+                        // TODO[JIRA REEF-1364]: Properly handle Exceptions.
+                        Logger.Log(Level.Error, "Received uncaught System Exception, force shutting down the Evaluator.");
```

Comment:
Using `InjectionFuture` doesn't work because it will try to create a new `EvaluatorRuntime` object, since `TaskRuntime` cannot be created before the fork the injector that creates the `EvaluatorRuntime`.